ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ClientConfigResourceProvider.java

```diff
@@ -441,6 +394,18 @@ public RequestStatus createResources(Request request)
           }
         }
 
+        TreeMap<String, String> clusterLevelParams = null;
+        TreeMap<String, String> ambariLevelParams = null;
+        if (getManagementController() instanceof AmbariManagementControllerImpl){
+          AmbariManagementControllerImpl controller = ((AmbariManagementControllerImpl)getManagementController());
+          clusterLevelParams = controller.getMetadataClusterLevelParams(cluster, stackId);
+          ambariLevelParams = controller.getMetadataAmbariLevelParams();
+        }
+        TreeMap<String, String> agentLevelParams = new TreeMap<>();
+        agentLevelParams.put("hostname", hostName);
+        agentLevelParams.put("public_hostname", publicHostName);
+        agentLevelParams.put("agentCacheDir", "/var/lib/ambari-agent/cache");
```

Comment:
Why is `/var/lib/ambari-agent/cache` hardcoded here? This seems like an Ambari agent-side configuration.